ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/StandaloneExecutor.java

```diff
@@ -115,36 +169,86 @@ private void showWelcomeMessage() {
   }
 
   private void executeStatements(final String queries) {
-    final List<QueryMetadata> queryMetadataList = ksqlEngine.createQueries(queries, ksqlConfig);
-    for (QueryMetadata queryMetadata : queryMetadataList) {
-      if (queryMetadata instanceof PersistentQueryMetadata) {
-        PersistentQueryMetadata persistentQueryMetadata = (PersistentQueryMetadata) queryMetadata;
-        persistentQueryMetadata.start();
-      } else {
+    final List<Pair<String, Statement>> statementPairs =
+        ksqlEngine.parseStatements(queries, ksqlEngine.getMetaStore().clone());
+    for (final Pair<String, Statement> statementPair: statementPairs) {
+      final String statementString = statementPair.getLeft();
+      final Statement statement = statementPair.getRight();
+      final BiConsumer<String, Statement> function = functionMap.get(statement.getClass());
+      if (function == null) {
         final String message = String.format(
             "Ignoring statements: %s"
-            + "%nOnly CREATE statements can run in standalone mode.",
-            queryMetadata.getStatementString()
+                + "%nOnly DDL (CREATE STREAM/TABLE, DROP STREAM/TABLE, SET, UNSET) "
+                + "and DML(CSAS, CTAS and INSERT INTO) statements can run in standalone mode.",
+            statementPair.getLeft()
         );
         System.err.println(message);
         log.warn(message);
       }
+      functionMap.get(statement.getClass()).accept(statementPair.getLeft(),
+          statementPair.getRight());
+    }
+
+  }
+
+
+  private void handlePersistentQuery(
+      final Statement statement,
+      final String statementString,
+      final Map<String, Object> configProperties) {
+    final Query query = getQueryFromStatement(statementString, statement);
+    final QueryMetadata queryMetadata =
+        ksqlEngine.getQueryExecutionPlan(query, ksqlConfig);
+    if (statement instanceof CreateAsSelect) {
```

Comment:
> @big-andy-coates 6 days ago  Member
> I think this if statement is not necessary (though could be wrong).

> Just rename validateCsasCtas to validate and change it to take a Statement...

>   @hjafarpour hjafarpour 3 days ago  Member
> It is needed since we can also have InsertInto as argument and the validation is only for ctas/csas.

The point is that the validate function already has if/else branching to only handle statements of a certain type. So you can do this:


```
private void handlePersistentQuery(
      final Statement statement,
      final String statementString,
      final Map<String, Object> configProperties) {
    final Query query = getQueryFromStatement(statementString, statement);
    final QueryMetadata queryMetadata =
        ksqlEngine.getQueryExecutionPlan(query, ksqlConfig);

    // -- Look mum, no if statement and no cast necessary:
    validate(queryMetadata, statement);

    final List<QueryMetadata> queryMetadataList =
        ksqlEngine.buildMultipleQueries(statementString, ksqlConfig, configProperties);
    if (queryMetadataList.size() != 1
        || !(queryMetadataList.get(0) instanceof PersistentQueryMetadata)) {
      throw new KsqlException("Could not build the query: " + statementString);
    }
    queryMetadataList.get(0).start();
  }

private void validate(final QueryMetadata queryMetadata, final Statement createAsSelect) {
    if (createAsSelect instanceof CreateStreamAsSelect
        && queryMetadata.getDataSourceType() == DataSource.DataSourceType.KTABLE) {
      throw new KsqlException("Invalid result type. Your SELECT query produces a TABLE. "
          + "Please use CREATE TABLE AS SELECT statement instead.");
    } else if (createAsSelect instanceof CreateTableAsSelect
        && queryMetadata.getDataSourceType() == DataSourceType.KSTREAM) {
      throw new KsqlException("Invalid result type. Your SELECT query produces a STREAM. "
          + "Please use CREATE STREAM AS SELECT statement instead.");
    }
  }
```

Reply:
Removed the `if` statement. After merging #1646 there won't be need for this validation.